src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,46 @@
+import { useParams } from "react-router-dom";
+import { useState } from "react";
+import { useEffect } from "react";
+import { getArticleById, getUserByUsername } from "../utils/api";
+
+function SingleArticle({isLoading, setIsLoading}) {
+    const { article_id } = useParams();
+    const [singleArticle, setSingleArticle] = useState({});
+    const [author, setAuthor] = useState({});
+
+    useEffect(() => {
+        setIsLoading(true)
+        getArticleById(article_id)
+            .then((articleFromApi) => {
+                setSingleArticle(articleFromApi)
+                getUserByUsername(articleFromApi.author)
+                    .then((author) => {
+                        setAuthor(author);
+                        setIsLoading(false);
+                    })
+            })
+    }, [article_id, setIsLoading])
+
+    if (isLoading) {
+        return (
+            <img src="https://img.pikbest.com/png-images/20190918/cartoon-snail-loading-loading-gif-animation_2734139.png!bw700" alt="loading" id="loading-img"/>
+        )
+    }
+
+    return (
+        <section className="single-article">
+            <h2>{singleArticle.title}</h2>
+            <div className="single-article-details">
+                <img id="author-avatar" src={author.avatar_url} alt={author.username} />
+                <p>{singleArticle.author}</p>
```

Comment:
I would use the author from the article here for the alt tag as if there is no image there quite possibly won't be a author.username either.